### ISSUE TITLE
[RN] Incompatibility with react-navigation

### ISSUE THREAD

**Issue by mxmzb** [algolia/react-instantsearch#1511]
**bug: using `react-instantsearch-native` with `react-navigation` and not having all the connected instantsearch widgets direct children of `InstantSearch` component.**

**Bug: What is the current behavior?**

i cannot put the search box (wrapped by `connectSearchBox()` hoc) into a custom header component that I pass to `navigationOptions.header` (https://reactnavigation.org/docs/en/stack-navigator.html#header). to connect that very search box i need to wrap it somewhere above in the tree with the `InstantSearch` (https://community.algolia.com/react-instantsearch/Getting_started_React_native.html#add-the-%3Cinstantsearch%3E-component) component somewhere above my first `react-navigation` component. however, if I do this, my app stops rendering anything and gives me a blank page. clicking anywhere results in https://github.com/facebook/react-devtools/issues/1032

tldr.: `InstantSearch` component seems to work only if the connected `react-instantsearch` components are direct children of it (or maybe there must be no `react-navigation` components between). i am currently forced to have the code from https://community.algolia.com/react-instantsearch/Getting_started_React_native.html more or less like in the examples in one screen and can not move the search box into `react-navigation` `navigationOptions.header` prop.

**Bug: What is the expected behavior?**

being able to have the search box and hits list in *very* (not sure myself _how_ "very") different components.

**Bug: What browsers are impacted? Which versions?**

`react-native` > 0.55.0, `react-navigation` > 2.0.0 and `react-instantsearch-native` = *

**Feature: What is your use case for such a feature?**

Having a search box properly in the header of a screen, because having it in the screen itself causes weird visual flaws (header unmounts, white background, whatsoever)

**What is the version you are using? Always use the latest one before opening a bug issue.**

`react-native` = 0.56.0, `react-navigation` = 2.11.2 and `react-instantsearch-native` = 5.2.3

**Comment by samouss** [comment]
I've setup an App with React Navigation & React InstantSearch. I've added the root above the `Router` but it's not mandatory if your Search is only on one screen. Here is [the source code](https://github.com/samouss/react-instantsearch-react-navigation) of the example. If the issue persist please provide us a working example of your App, it will help a lot to better understand where the issue come from. Thanks!

**Comment by samouss** [comment]
I'm closing this issue because it's inactive since a while. Feel free to open a new one!
